### PR TITLE
add blanks, hints and the test codes

### PR DESCRIPTION
**Pull request by Zhaochengxiann**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [ ] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <p>Is this needed?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #24.            from pandas.testing import assert_frame_equal
</pre><p>move the import to the beginning of this cell.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #25.    
</pre><p>remove the unused empty line.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <p>You can have some questions here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    def get_survival_rate_by_gender(gender):
</pre><p>Please add some documentation for this function.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    def get_Histogram_of_AgeFill_segmented_by_Survived(df, column_number):
</pre><p> </p><p> </p><p>def get_age_by_survived(df, survived):</p><p> </p><p> </p><p>And please add documentation and questions for this function.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    def get_the_unique_values_of_Pclass(df):
</pre><p>Please add some documentation for this function. And don't have upper case for python method.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.            df.AgeFill[df.Pclass == pclass].____(kind="kde")
</pre><p>better to use df["AgeFill"]?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: no, i forgot to remove it

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: ok, done

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: done

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <p>ok</p>

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <p>done</p>

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <p>done</p>

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <p>yeah</p>

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <p>ok</p>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.        Args:
</pre><p>If your function have return values, you should include it in the documentation as well.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    def get_age_by_survived(df, survived):
</pre><p>same here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.        """plot the column 'AgeFill' for each unique value of 'Pclass' in the input DataFrame..
</pre><p>plot -&gt; Plot</p><p> </p><p>of_Pclass -&gt; of_plcass</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: ok

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: ok

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:1]: thanks

**Review by zhangqi444 — APPROVED** [review]

**Pull request merged**